bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -413,39 +425,44 @@ public void writeValue(String variableName, String value) throws HeliosException
+                        // write to device
+                        String payload = v.getVariableString() + "=" + value;
+                        ModbusCommunicationInterface comms = this.comms;
+                        if (comms != null) {
+                            final Semaphore lock = transactionLocks.get(comms.getEndpoint());
+                            try {
+                                if (lock != null) {
+                                    lock.acquire();
+                                    comms.submitOneTimeWrite(new ModbusWriteRegisterRequestBlueprint(
+                                            HeliosEasyControlsBindingConstants.UNIT_ID,
                                             HeliosEasyControlsBindingConstants.START_ADDRESS, preparePayload(payload),
-                                            true, HeliosEasyControlsBindingConstants.MAX_TRIES),
-                                    result -> {
-                                        lock.release();
-                                        updateStatus(ThingStatus.ONLINE);
-                                    }, failureInfo -> {
-                                        lock.release();
-                                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                                                "Error writing to device: " + failureInfo.getCause().getMessage());
-                                    });
-                        } catch (InterruptedException e) {
-                            logger.warn(
-                                    "{} encountered Exception when trying to lock Semaphore for writing variable {} to the device: {}",
-                                    HeliosEasyControlsHandler.class.getSimpleName(), variableName, e.getMessage());
+                                            true, HeliosEasyControlsBindingConstants.MAX_TRIES), result -> {
+                                                lock.release();
+                                                updateStatus(ThingStatus.ONLINE);
+                                            }, failureInfo -> {
+                                                lock.release();
+                                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                                        "Error writing to device: "
+                                                                + failureInfo.getCause().getMessage());
+                                            });
+                                }
+                            } catch (InterruptedException e) {
+                                logger.warn(
+                                        "{} encountered Exception when trying to lock Semaphore for writing variable {} to the device: {}",
+                                        HeliosEasyControlsHandler.class.getSimpleName(), variableName, e.getMessage());
```

Comment:
This exception is thrown when OH is shut-down. That means the current operation should be finished as fast as possible and no further operations should be done. You can achieve this by not catching the exception here but throw the InterruptedException and catch it at the highest level you can. So, this is not an error if this exception is raised.

Reply:
Thanks for the input! I hope the changes I made are what you had in mind.